crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -0,0 +1,348 @@
+            target: cow_amm.address(),
+            value: 0.into(),
+            calldata: commitment.0.to_vec(),
+        }
+    };
+
+    // fund trader "bob" and approve vault relayer
+    tx_value!(
+        bob.account(),
+        U256::exp10(17),
+        onchain.contracts().weth.deposit()
+    );
+    tx!(
+        bob.account(),
+        onchain
+            .contracts()
+            .weth
+            .approve(onchain.contracts().allowance, U256::MAX)
+    );
+
+    // place user order with the same limit price as the CoW AMM order
+    let user_order = OrderCreation {
+        sell_token: onchain.contracts().weth.address(),
+        sell_amount: U256::exp10(17), // 0.1 WETH
+        buy_token: dai.address(),
+        buy_amount: to_wei(230), // 230 DAI
+        valid_to: model::time::now_in_epoch_seconds() + 300,
+        kind: OrderKind::Sell,
+        ..Default::default()
+    }
+    .sign(
+        EcdsaSigningScheme::Eip712,
+        &onchain.contracts().domain_separator,
+        SecretKeyRef::from(&SecretKey::from_slice(bob.private_key()).unwrap()),
+    );
+    let user_order_id = services.create_order(&user_order).await.unwrap();
+
+    let amm_balance_before = dai.balance_of(cow_amm.address()).call().await.unwrap();
+    let bob_balance_before = dai.balance_of(bob.address()).call().await.unwrap();
+
+    mock_solver.configure_solution(Some(Solution {
+        id: 1,
+        // assume price of the univ2 pool
+        prices: HashMap::from([
+            (dai.address(), to_wei(100)),
+            (onchain.contracts().weth.address(), to_wei(300_000)),
+        ]),
+        trades: vec![
+            solvers_dto::solution::Trade::Jit(solvers_dto::solution::JitTrade {
+                order: solvers_dto::solution::JitOrder {
+                    sell_token: cow_amm_order.sell_token,
+                    buy_token: cow_amm_order.buy_token,
+                    receiver: cow_amm_order.receiver.unwrap_or_default(),
+                    sell_amount: cow_amm_order.sell_amount,
+                    buy_amount: cow_amm_order.buy_amount,
+                    valid_to: cow_amm_order.valid_to,
+                    app_data: cow_amm_order.app_data.0,
+                    fee_amount: cow_amm_order.fee_amount,
+                    kind: Kind::Sell,
+                    partially_fillable: cow_amm_order.partially_fillable,
+                    sell_token_balance: SellTokenBalance::Erc20,
+                    buy_token_balance: BuyTokenBalance::Erc20,
+                    signing_scheme: SigningScheme::Eip1271,
+                    signature,
+                },
+                executed_amount: cow_amm_order.sell_amount,
+            }),
+            solvers_dto::solution::Trade::Fulfillment(solvers_dto::solution::Fulfillment {
+                executed_amount: user_order.sell_amount,
+                fee: Some(0.into()),
+                order: user_order_id.0,
+            }),
+        ],
+        pre_interactions: vec![cow_amm_commitment],
```

Comment:
Why do we need this here if in [this](https://github.com/cowprotocol/services/pull/2766/files#diff-4c8f610bdbea68810a9395f71782c8a9fb5a6931e9e814a2f5f9b33bdc847e2eR101) line we remove the `pre_interactions` from the JIT order?

Reply:
Couldn't find the reference quickly but initially we had implemented that solvers could make these interactions part of the JIT order pre-interactions themselves. But Felix suggested that instead it would also be sufficient to have 1 list of pre-interactions attached to the whole settlement instead of at each individual order.
If we instead had kept it such that solvers provide pre-interactions on individual orders we clone the interactions in the line you linked instead of passing it here.